pkg/neonvm/controllers/vm_controller.go

```diff
@@ -864,12 +865,45 @@ func runnerStatus(pod *corev1.Pod) runnerStatusKind {
 	case corev1.PodFailed:
 		return runnerFailed
 	case corev1.PodRunning:
-		return runnerRunning
+		return isRunnerPodReady(pod)
 	default:
 		panic(fmt.Errorf("unknown pod phase: %q", pod.Status.Phase))
 	}
 }
 
+// isRunnerPodReady returns whether the runner pod is ready respecting the readiness probe of its containers.
+func isRunnerPodReady(pod *corev1.Pod) runnerStatusKind {
+	// if the pod has no container statuses, we consider it pending
+	if len(pod.Status.ContainerStatuses) == 0 {
+		return runnerPending
+	}
+
+	// first we check if the pod is a target pod for the migration
+	// in that case we don't wait for the readiness probe
+	// from neonvm-daemon because qemu started
+	// in incoming migration mode
+	for _, containerSpec := range pod.Spec.Containers {
+		if containerSpec.Name == "neonvm-runner" {
+			for _, env := range containerSpec.Env {
+				if env.Name == "RECEIVE_MIGRATION" && env.Value == "true" {
+					return runnerRunning
```

Comment:
question(blocking): Does this mean that we'll return `runnerRunning` even when the neonvm-runner container is not yet running? (or alternately: can this return ready even when the pod is still pending?)

Reply:
Fair point. I introduce the new status and change the vmmigration controller to use it instead of running.

Reply:
After the discussion with you and @Omrigan I changed the code here to check the pod references to determine if it is a target pod for migration, which looks like the most reliable way we can currently afford.